core/dbt/task/rpc/cli.py

```diff
@@ -36,6 +37,11 @@ def __init__(self, args, config, manifest):
 
     def set_config(self, config):
         super().set_config(config)
+
+        # patch up config with args/cli_vars
+        self.config.args = self.args
+        self.config.cli_vars = parse_cli_vars(getattr(self.args, 'vars', '{}'))
```

Comment:
Can we update the request-supplied `--vars` on top of any vars supplied to `dbt rpc`? I like the idea of the rpc-runner being able to specify base variables that are then updated in the scope of a particular request.

If this is onerous or anything other than a 1-line change to this PR, then we should discuss it further!

Reply:
I think we'd also want to patch up `self.args.vars`, but that should be no big deal.